File: src/protocol/BlindsUPCommand.cpp
#include "BlindsUPCommand.h"
#include "CommandVisitor.h"
#include <utility/BytesConverter.h>

#include <stdexcept>

using namespace std;
using namespace utility;
using namespace communication;

BlindsUPCommand::BlindsUPCommand()
    : Command(10,CommandType::BLINDS_UP)
{}

BlindsUPCommand::~BlindsUPCommand()
{}

vector<uint8_t> BlindsUPCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

void BlindsUPCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}

string BlindsUPCommand::getName()
{
    return string("BlindsUPCommand");
}

void BlindsUPCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/BlindsUPCommand.java
package wpam.mobile_client.protocol;

import java.util.ArrayList;

public class BlindsUPCommand extends Command
	{
		public BlindsUPCommand()
		{
			super(10, CommandType.BLINDS_UP);
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(commandType_.getValue());

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("BlindsUPCommand");
		}

		@Override
		public void accept(CommandVisitor visitor)
		{
			visitor.visit(this);
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(1);
		}
	}